kernel/poll.c

```diff
@@ -213,6 +213,8 @@ static inline int register_events(struct k_poll_event *events,
 		} else if (!just_check && poller->is_polling) {
 			register_event(&events[ii], poller);
 			events_registered += 1;
+		} else {
+			__ASSERT(false, "Unknown condition cannot register events.");
```

Comment:
Looks like this is problematic and not appropriate assertion here, similar to work.c. I'll fix it here and below.